src/components/SplitButton/SplitButton.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { IMenuItemProps, Menu, MenuItem, Position } from '@blueprintjs/core'
+import cx from 'classnames'
+import React, { MouseEvent } from 'react'
+import { Button, ButtonProps, Popover } from '../..'
+import css from './SplitButton.css'
+
+type SplitButtonProps = Omit<ButtonProps, 'rightIcon'> & { className?: string }
+
+export const SplitButton: React.FC<SplitButtonProps> = ({
+  onClick,
+  text,
+  icon,
+  children,
+  className,
+  ...commonProps
+}) => {
+  const [isOptionsOpen, setOptionsOpen] = React.useState(false)
+
+  const handleClick = async (event: MouseEvent) => {
+    onClick?.(event)
+  }
+
+  return (
+    <div className={css.splitButton}>
+      <Button {...commonProps} onClick={handleClick} text={text} icon={icon} className={cx(css.main)} />
+      <Popover
+        isOpen={isOptionsOpen}
+        onInteraction={nextOpenState => {
+          setOptionsOpen(nextOpenState)
+        }}
+        content={<Menu>{children}</Menu>}
+        usePortal={false}
+        minimal={true}
+        fill={false}
+        position={Position.BOTTOM_RIGHT}>
+        <Button
+          rightIcon="chevron-down"
```

Comment:
If we used icon here instead of rightIcon, would it help in shifting the icon a little to the left and centre it?

Reply:
that again causes more padding towards the right. 

<img width="50" alt="image" src="https://user-images.githubusercontent.com/20707504/171795760-38ef954c-3156-4961-998e-09e58bd0ce62.png">

Let me override the blueprint CSS to have common horizontal padding